### ISSUE TITLE
Paths level Parameters are ignored

### ISSUE THREAD

**Issue by MichaelMay81** [Zaid-Ajaj/Hawaii#31]
I have to parse a json which has parameters on Path level.

I did a quick fix by extending ```operationParameters``` and calling it in ```createOpenApiClient``` like this:
```
let parameters = operationParameters operationInfo pathInfo.Parameters visitedTypes config
```
pathInfo.Parameters are concated with the operation.Parameters. I haven't checked yet if that does the overriding like specified in the OpenAPI documentation.

Does that sound like a valid PullRequest, or am I missing something? :)

**Comment by Zaid-Ajaj** [comment]
> I have to parse a json which has parameters on Path level.

Hi @MichaelMay81 can you please elaborate? I will need an example schema, what Hawaii generates and what it should have generated to get a better picture of the issue

**Comment by MichaelMay81** [comment]
```
"paths": {
    "/repositories/{repository}/trees": {
      "parameters": [
        { "name": "repository",
          "in": "path",
          "required": true,
          "schema": {"type": "string"}
       }],
      "get": {
        "operationId": "GetTrees",
        "parameters": [
          {
            "name": "name",
            "in": "query",
            "required": false,
            "schema": {"type": "string"}
          },
          { "name": "description",
            "in": "query",
            "required": false,
            "schema": {"type": "string" }
          }
```
generates:
```member this.GetTrees(?name: string, ?description: string)```
but should generate:
```member this.GetTrees(repository: string, ?name: string, ?description: string)```

**Comment by Zaid-Ajaj** [comment]
Thanks for the clarification @MichaelMay81, I see the problem now. The parameters should be indeed merged as you suggested. If you want to give a try and send a PR, would really appreciate it 🙏

**Comment by MichaelMay81** [comment]
Made a quick and dirty fix by adding a parameter for ```pathInfo.Parameters``` to ```operationParameters```, but would rather do an overhaul of that function.
I am thinking of following points:
- Remove ```visitedTypes``` because it is not used
- Split function ```operationParameters``` into smaller functions, e.g., ```processOperationParameters``` and ```processOperationRequestBody```
- Swap out ```ResizeArray``` for ```list``` for a more functional style
- Move that functionality into OperationParameters.fs

What do you think?

**Comment by Zaid-Ajaj** [comment]
I like the proposal @MichaelMay81 💯 sounds like a good plan

**Comment by Zaid-Ajaj** [comment]
Resolved by #34
